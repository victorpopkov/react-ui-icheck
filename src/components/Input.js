import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const iCheck = 'iCheck';
const iCheckHelper = `${iCheck}-helper`;

class Input extends Component {
  static propTypes = {
    inputType: PropTypes.string.isRequired,
    activeClass: PropTypes.string,
    activeClassName: PropTypes.string,
    aria: PropTypes.bool,
    checkboxClass: PropTypes.string,
    checked: PropTypes.bool,
    checkedClass: PropTypes.string,
    className: PropTypes.string,
    defaultChecked: PropTypes.bool,
    disabled: PropTypes.bool,
    disabledClass: PropTypes.string,
    focusClass: PropTypes.string,
    hoverClass: PropTypes.string,
    hovered: PropTypes.bool,
    id: PropTypes.string,
    increaseArea: PropTypes.string,
    indeterminate: PropTypes.bool,
    indeterminateClass: PropTypes.string,
    inheritClass: PropTypes.bool,
    inheritID: PropTypes.bool,
    insert: PropTypes.node,
    isMobile: PropTypes.bool,
    label: PropTypes.string,
    name: PropTypes.string,
    radioClass: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
  };

  static defaultProps = {
    activeClass: null,
    activeClassName: null,
    aria: false,
    checkboxClass: 'icheckbox',
    checked: false,
    checkedClass: 'checked',
    className: null,
    defaultChecked: false,
    disabled: false,
    disabledClass: 'disabled',
    focusClass: 'focus',
    hoverClass: 'hover',
    hovered: false,
    id: null,
    increaseArea: '',
    indeterminate: false,
    indeterminateClass: 'indeterminate',
    inheritClass: false,
    inheritID: false,
    insert: null,
    isMobile: false,
    label: null,
    name: null,
    radioClass: 'iradio',
    value: '',
    onBlur: null,
    onChange: null,
    onFocus: null,
  };

  constructor(props) {
    super(props);

    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleHelperClick = this.handleHelperClick.bind(this);

    const {
      checked,
      defaultChecked,
      hovered,
      indeterminate,
    } = props;

    this.state = {
      active: false,
      checked: defaultChecked || checked,
      focused: false,
      hovered,
      indeterminate,
    };
  }

  componentWillReceiveProps(nextProps) {
    const { checked: prevChecked, hovered: prevHovered } = this.props;
    const { checked, hovered } = nextProps;

    if (checked !== prevChecked) {
      this.setState({
        checked,
      });
    }

    if (hovered !== prevHovered) {
      this.setState({
        hovered,
      });
    }
  }

  handleBlur(event) {
    const { onBlur } = this.props;

    this.setState({
      focused: false,
    });

    if (onBlur) {
      onBlur(event);
    }
  }

  handleChange(event) {
    const { onChange } = this.props;
    const { checked } = event.target;

    this.setState(state => ({
      checked: !state.checked,
    }));

    if (onChange) {
      onChange(event, checked);
    }
  }

  handleFocus(event) {
    const { onFocus } = this.props;

    this.setState({
      focused: true,
    });

    if (onFocus) {
      onFocus(event);
    }
  }

  handleHelperClick(event) {
    const {
      disabled,
      inputType,
      insert,
      label,
      onChange,
    } = this.props;
    const { checked } = this.state;

    if ((!label && !insert) || disabled) {
      return;
    }

    if (inputType === 'checkbox') {
      this.setState(state => ({
        checked: !state.checked,
      }));
    }

    event.preventDefault();
    event.stopPropagation();

    if (onChange) {
      // make sure <ins /> element is not target
      event.target = this.checkbox; // eslint-disable-line no-param-reassign
      onChange(event, checked);
    }
  }

  render() {
    const {
      activeClass,
      activeClassName,
      aria,
      checkboxClass,
      checkedClass,
      className,
      disabled,
      disabledClass,
      focusClass,
      hoverClass,
      id,
      increaseArea,
      indeterminateClass,
      inheritClass,
      inheritID,
      inputType,
      insert,
      isMobile,
      name,
      radioClass,
      value,

      defaultChecked,
      hovered,
      indeterminate,

      ...other
    } = this.props;

    const {
      active,
      checked,
      focused,
      hovered: stateHovered,
      indeterminate: stateIndeterminate,
    } = this.state;

    const wrapProps = {
      className: classNames({
        [checkboxClass]: inputType === 'checkbox',
        [radioClass]: inputType === 'radio',
        [indeterminateClass]: stateIndeterminate,
        [checkedClass]: checked,
        [disabledClass]: disabled,
        [hoverClass]: stateHovered,
        [focusClass]: focused,
        [activeClassName || activeClass]: active,
      }),
    };

    if (inheritClass) {
      wrapProps.className = classNames(wrapProps.className, className);
    }

    if (inheritID && id) {
      wrapProps.id = `${iCheck}-${id}`;
    }

    if (aria) {
      wrapProps.role = inputType;
      // Set ARIA "labelledby"
      wrapProps['aria-labelledby'] = wrapProps.id;
      wrapProps['aria-checked'] = checked;
    }

    // Clickable area
    let area = (`${increaseArea}`).replace('%', '') || 0;
    if (area < -50) {
      area = -50;
    }

    // Layer styles
    const offset = `${-area}%`;
    const size = `${100 + (area * 2)}%`;
    const layer = {
      position: 'absolute',
      top: offset,
      left: offset,
      display: 'block',
      width: size,
      height: size,
      margin: 0,
      padding: 0,
      background: '#fff',
      border: 0,
      opacity: 0,
      cursor: disabled ? 'default' : 'pointer',
    };

    // Layer addition
    const helper = (
      <ins
        className={iCheckHelper}
        role="presentation"
        style={layer}
        onClick={this.handleHelperClick}
      />
    );

    // Choose how to hide input
    let hide;
    if (isMobile) {
      hide = {
        position: 'absolute',
        visibility: 'hidden',
      };
    } else if (area) {
      hide = layer;
    } else {
      hide = {
        opacity: 0,
        position: 'absolute',
      };
    }

    const inputElement = (
      <input
        {...other}
        checked={checked}
        disabled={disabled}
        name={name}
        ref={(c) => { this.checkbox = c; }}
        style={hide}
        type={inputType}
        value={value}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
        onFocus={this.handleFocus}
      />
    );

    let insertElement = null;
    if (insert) {
      insertElement = React.isValidElement(insert)
        ? insert
        : (
          // eslint-disable-next-line react/no-danger
          <div dangerouslySetInnerHTML={{ __html: insert }} />
        );
    }

    return (
      <div {...wrapProps}>
        {inputElement}
        {insertElement}
        {helper}
      </div>
    );
  }
}

export default Input;
