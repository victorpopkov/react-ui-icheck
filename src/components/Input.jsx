import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const iCheck = 'iCheck';
const iCheckHelper = `${iCheck}-helper`;

class Input extends Component {
  constructor(props) {
    super(props);

    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleHelperClick = this.handleHelperClick.bind(this);

    const { checked, defaultChecked, hovered, indeterminate } = props;

    this.state = {
      active: false,
      checked: defaultChecked || checked,
      focused: false,
      prevChecked: null,
      prevHovered: null,
      hovered,
      indeterminate,
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (
      state.prevChecked !== props.checked &&
      props.checked !== state.checked
    ) {
      return {
        checked: props.defaultChecked || props.checked,
        prevChecked: state.prevChecked,
      };
    }

    if (
      state.prevHovered !== props.hovered &&
      props.hovered !== state.hovered
    ) {
      return {
        hovered: props.hovered,
        prevHovered: state.prevHovered,
      };
    }

    return {
      prevChecked: props.checked,
      prevHovered: props.hovered,
    };
  }

  handleBlur(event) {
    const { onBlur } = this.props;

    this.setState({ focused: false });

    if (onBlur) onBlur(event);
  }

  handleChange(event) {
    const { onChange } = this.props;
    const { checked } = event.target;

    this.setState((state) => ({ checked: !state.checked }));

    if (onChange) onChange(event, checked);
  }

  handleFocus(event) {
    const { onFocus } = this.props;

    this.setState({ focused: true });

    if (onFocus) onFocus(event);
  }

  handleHelperClick(event) {
    const { disabled, inputType, insert, label, onChange } = this.props;
    const { checked } = this.state;

    if ((!label && !insert) || disabled) return;

    if (inputType.toString() === 'checkbox')
      this.setState((state) => ({ checked: !state.checked }));

    event.preventDefault();
    event.stopPropagation();

    if (onChange) {
      // make sure <ins /> element is not target
      event.target = this.checkbox; // eslint-disable-line no-param-reassign
      onChange(event, checked);
    }
  }

  render() {
    const {
      activeClassName,
      aria,
      checkboxClassName,
      checkedClassName,
      className,
      disabled,
      disabledClassName,
      focusClassName,
      hoverClassName,
      id,
      increaseArea,
      indeterminateClassName,
      inheritClassName,
      inheritID,
      inputType,
      insert,
      isMobile,
      name,
      radioClassName,
      value,

      defaultChecked,
      hovered,
      indeterminate,

      ...other
    } = this.props;

    const {
      active,
      checked,
      focused,
      hovered: stateHovered,
      indeterminate: stateIndeterminate,
    } = this.state;

    const wrapProps = {
      className: classNames({
        [checkboxClassName]: inputType.toString() === 'checkbox',
        [radioClassName]: inputType.toString() === 'radio',
        [indeterminateClassName]: stateIndeterminate,
        [checkedClassName]: checked,
        [disabledClassName]: disabled,
        [hoverClassName]: stateHovered,
        [focusClassName]: focused,
        [activeClassName]: active,
      }),
    };

    if (inheritClassName)
      wrapProps.className = classNames(wrapProps.className, className);

    if (inheritID && id) wrapProps.id = `${iCheck}-${id}`;

    if (aria) {
      wrapProps.role = inputType;
      // Set ARIA "labelledby"
      wrapProps['aria-labelledby'] = wrapProps.id;
      wrapProps['aria-checked'] = checked;
    }

    // Clickable area
    let area = `${increaseArea}`.replace('%', '') || 0;
    if (area < -50) area = -50;

    // Layer styles
    const offset = `${-area}%`;
    const size = `${100 + area * 2}%`;
    const layer = {
      position: 'absolute',
      top: offset,
      left: offset,
      display: 'block',
      width: size,
      height: size,
      margin: 0,
      padding: 0,
      background: '#fff',
      border: 0,
      opacity: 0,
      cursor: disabled ? 'default' : 'pointer',
    };

    // Layer addition
    const helper = (
      <ins
        className={iCheckHelper}
        role="presentation"
        style={layer}
        onClick={this.handleHelperClick}
      />
    );

    // Choose how to hide input
    let hide;
    if (isMobile) {
      hide = {
        position: 'absolute',
        visibility: 'hidden',
      };
    } else if (area) {
      hide = layer;
    } else {
      hide = {
        opacity: 0,
        position: 'absolute',
      };
    }

    const inputElement = (
      <input
        {...other}
        checked={checked}
        disabled={disabled}
        name={name}
        ref={(c) => {
          this.checkbox = c;
        }}
        style={hide}
        type={inputType}
        value={value}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
        onFocus={this.handleFocus}
      />
    );

    let insertElement = null;
    if (insert) {
      insertElement = React.isValidElement(insert) ? (
        insert
      ) : (
        // eslint-disable-next-line react/no-danger
        <div dangerouslySetInnerHTML={{ __html: insert }} />
      );
    }

    return (
      <div {...wrapProps}>
        {inputElement}
        {insertElement}
        {helper}
      </div>
    );
  }
}

Input.propTypes = {
  inputType: PropTypes.string.isRequired,
  activeClassName: PropTypes.string,
  aria: PropTypes.bool,
  checkboxClassName: PropTypes.string,
  checked: PropTypes.bool,
  checkedClassName: PropTypes.string,
  className: PropTypes.string,
  defaultChecked: PropTypes.bool,
  disabled: PropTypes.bool,
  disabledClassName: PropTypes.string,
  focusClassName: PropTypes.string,
  hoverClassName: PropTypes.string,
  hovered: PropTypes.bool,
  id: PropTypes.string,
  increaseArea: PropTypes.string,
  indeterminate: PropTypes.bool,
  indeterminateClassName: PropTypes.string,
  inheritClassName: PropTypes.bool,
  inheritID: PropTypes.bool,
  insert: PropTypes.node,
  isMobile: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string,
  radioClassName: PropTypes.string,
  value: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
};

Input.defaultProps = {
  activeClassName: null,
  aria: false,
  checkboxClassName: 'icheckbox',
  checked: false,
  checkedClassName: 'checked',
  className: null,
  defaultChecked: false,
  disabled: false,
  disabledClassName: 'disabled',
  focusClassName: 'focus',
  hoverClassName: 'hover',
  hovered: false,
  id: null,
  increaseArea: '',
  indeterminate: false,
  indeterminateClassName: 'indeterminate',
  inheritClassName: false,
  inheritID: false,
  insert: null,
  isMobile: false,
  label: null,
  name: null,
  radioClassName: 'iradio',
  value: '',
  onBlur: null,
  onChange: null,
  onFocus: null,
};

export default Input;
