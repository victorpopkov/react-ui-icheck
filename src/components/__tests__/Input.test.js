import { mount, render, shallow } from 'enzyme';
import React from 'react';
import expect from 'expect';
import Input from '../Input'; // eslint-disable-line sort-imports

const testInputComponent = (inputType) => {
  const testInput = <Input inputType={inputType} />;
  const testSpy = jest.spyOn(Input.prototype, 'componentWillReceiveProps');

  afterEach(() => {
    testSpy.mockClear();
  });

  it('should have the default props values when mounted', () => {
    const wrapper = mount(testInput);
    expect(wrapper.props()).toMatchSnapshot();
  });

  it('should render the corresponding HTML when the insert value is set', () => {
    const input = (
      <Input
        inputType={inputType}
        insert={(
          <div>test</div>
        )}
      />
    );

    const wrapper = render(input);
    expect(wrapper).toMatchSnapshot();
  });

  describe('should render wrapper', () => {
    it('with corresponding className when inheritClass is passed', () => {
      const wrapper = shallow(<Input className="test" inputType={inputType} inheritClass />);
      expect(wrapper.props().className).toBe(`i${inputType} test`);
    });

    it('with corresponding id when inheritID is passed', () => {
      const wrapper = shallow(<Input id="test" inputType={inputType} inheritID />);
      expect(wrapper.props().id).toBe('iCheck-test');
    });

    describe('with aria labels ', () => {
      const testAriaInput = <Input inputType={inputType} aria />;

      it('when aria prop is passed', () => {
        const wrapper = shallow(testAriaInput);
        expect(wrapper).toMatchSnapshot();
      });

      it('when aria prop is passed and the checked state changes', () => {
        const wrapper = shallow(testAriaInput);

        expect(wrapper.props()['aria-checked']).toBe(false);
        wrapper.setState({ checked: true });
        expect(wrapper.props()['aria-checked']).toBe(true);
      });
    });
  });

  describe('should render the <input />', () => {
    it('on mobiles', () => {
      const wrapper = shallow(testInput);
      wrapper.setProps({ isMobile: true });
      expect(wrapper.find('input').prop('style')).toMatchSnapshot();
    });

    describe('when increaseArea is passed', () => {
      it('as a positive value', () => {
        const wrapper = shallow(<Input increaseArea="30%" inputType={inputType} />);
        expect(wrapper).toMatchSnapshot();
      });

      it('as a negative value', () => {
        const wrapper = shallow(<Input increaseArea="-30%" inputType={inputType} />);
        expect(wrapper.find('input').props()).toMatchSnapshot();
      });

      it('as a negative value and more than 50%', () => {
        const wrapper = shallow(<Input increaseArea="-60%" inputType={inputType} />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('should update the state', () => {
    it('when the checked prop has changed', () => {
      const wrapper = mount(testInput);

      expect(testSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.props().checked).toBe(false);
      expect(wrapper.state().checked).toBe(false);

      wrapper.setProps({ checked: true });

      expect(testSpy).toHaveBeenCalledTimes(1);
      expect(wrapper.props().checked).toBe(true);
      expect(wrapper.state().checked).toBe(true);
    });

    it('when the hovered prop has changed', () => {
      const wrapper = mount(testInput);

      expect(testSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.props().hovered).toBe(false);
      expect(wrapper.state().hovered).toBe(false);

      wrapper.setProps({ hovered: true });

      expect(testSpy).toHaveBeenCalledTimes(1);
      expect(wrapper.props().hovered).toBe(true);
      expect(wrapper.state().hovered).toBe(true);
    });
  });

  describe('should handle the <input />', () => {
    it('onFocus()', () => {
      const spy = jest.fn();
      const wrapper = mount(<Input inputType={inputType} onFocus={spy} />);

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().focused).toBe(false);

      wrapper.find('input').simulate('focus');

      expect(spy).toHaveBeenCalledTimes(1);
      expect(wrapper.state().focused).toBe(true);
    });

    it('onBlur()', () => {
      const spy = jest.fn();
      const wrapper = mount(<Input inputType={inputType} onBlur={spy} />);

      wrapper.setState({ focused: true });

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().focused).toBe(true);

      wrapper.find('input').simulate('blur');

      expect(spy).toHaveBeenCalledTimes(1);
      expect(wrapper.state().focused).toBe(false);
    });

    it('onChange()', () => {
      const spy = jest.fn();
      const wrapper = mount(<Input inputType={inputType} onChange={spy} />);

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().checked).toBe(false);

      wrapper.find('input').simulate('change');

      expect(spy).toHaveBeenCalledTimes(1);
      expect(wrapper.state().checked).toBe(true);
    });
  });

  describe('should handle the <ins />', () => {
    it('onClick()', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <Input
          inputType={inputType}
          insert="Insert"
          label="Label"
          onChange={spy}
        />
      );

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().checked).toBe(false);

      wrapper.find('ins').simulate('click');

      expect(spy).toHaveBeenCalledTimes(1);

      if (inputType === 'checkbox') {
        expect(wrapper.state().checked).toBe(true);
      } else {
        expect(wrapper.state().checked).toBe(false);
      }
    });

    it('onClick() when both label and insert are not set', () => {
      const wrapper = mount(testInput);

      expect(wrapper.state().checked).toBe(false);
      wrapper.find('ins').simulate('click');
      expect(wrapper.state().checked).toBe(false);
    });

    it('onClick() when disabled', () => {
      const wrapper = mount(<Input inputType={inputType} disabled />);

      expect(wrapper.state().checked).toBe(false);
      wrapper.find('ins').simulate('click');
      expect(wrapper.state().checked).toBe(false);
    });
  });
};

describe('<Input /> component', () => {
  describe('with the prop inputType set to "checkbox"', () => {
    testInputComponent('checkbox');
  });

  describe('with the prop inputType set to "radio"', () => {
    testInputComponent('radio');
  });
});
