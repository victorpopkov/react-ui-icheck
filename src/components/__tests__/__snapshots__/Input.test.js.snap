// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component with the prop inputType set to "checkbox" should have the default props values when mounted 1`] = `
Object {
  "activeClass": null,
  "activeClassName": null,
  "aria": false,
  "checkboxClass": "icheckbox",
  "checkboxClassName": "icheckbox",
  "checked": false,
  "checkedClass": "checked",
  "checkedClassName": "checked",
  "className": null,
  "defaultChecked": false,
  "disabled": false,
  "disabledClass": "disabled",
  "disabledClassName": "disabled",
  "focusClass": "focus",
  "hoverClass": "hover",
  "hovered": false,
  "id": null,
  "increaseArea": "",
  "indeterminate": false,
  "indeterminateClass": "indeterminate",
  "inheritClass": false,
  "inheritID": false,
  "inputType": "checkbox",
  "insert": null,
  "isMobile": false,
  "label": null,
  "name": null,
  "onBlur": null,
  "onChange": null,
  "onFocus": null,
  "radioClass": "iradio",
  "value": "",
}
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render the <input /> on mobiles 1`] = `
Object {
  "position": "absolute",
  "visibility": "hidden",
}
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render the <input /> when increaseArea is passed as a negative value 1`] = `
Object {
  "checked": false,
  "disabled": false,
  "label": null,
  "name": null,
  "onBlur": [Function],
  "onChange": [Function],
  "onFocus": [Function],
  "style": Object {
    "background": "#fff",
    "border": 0,
    "cursor": "pointer",
    "display": "block",
    "height": "40%",
    "left": "30%",
    "margin": 0,
    "opacity": 0,
    "padding": 0,
    "position": "absolute",
    "top": "30%",
    "width": "40%",
  },
  "type": "checkbox",
  "value": "",
}
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render the <input /> when increaseArea is passed as a negative value and more than 50% 1`] = `
<div
  className="icheckbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render the <input /> when increaseArea is passed as a positive value 1`] = `
<div
  className="icheckbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render the corresponding HTML when the insert value is set 1`] = `
<div
  class="icheckbox"
>
  <input
    style="opacity:0;position:absolute"
    type="checkbox"
    value=""
  />
  <div>
    test
  </div>
  <ins
    class="iCheck-helper"
    role="presentation"
    style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "checkbox" should render wrapper with aria labels  when aria prop is passed 1`] = `
<div
  aria-checked={false}
  className="icheckbox"
  role="checkbox"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
    type="checkbox"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "100%",
        "left": "0%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "0%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "radio" should have the default props values when mounted 1`] = `
Object {
  "activeClass": null,
  "activeClassName": null,
  "aria": false,
  "checkboxClass": "icheckbox",
  "checkboxClassName": "icheckbox",
  "checked": false,
  "checkedClass": "checked",
  "checkedClassName": "checked",
  "className": null,
  "defaultChecked": false,
  "disabled": false,
  "disabledClass": "disabled",
  "disabledClassName": "disabled",
  "focusClass": "focus",
  "hoverClass": "hover",
  "hovered": false,
  "id": null,
  "increaseArea": "",
  "indeterminate": false,
  "indeterminateClass": "indeterminate",
  "inheritClass": false,
  "inheritID": false,
  "inputType": "radio",
  "insert": null,
  "isMobile": false,
  "label": null,
  "name": null,
  "onBlur": null,
  "onChange": null,
  "onFocus": null,
  "radioClass": "iradio",
  "value": "",
}
`;

exports[`<Input /> component with the prop inputType set to "radio" should render the <input /> on mobiles 1`] = `
Object {
  "position": "absolute",
  "visibility": "hidden",
}
`;

exports[`<Input /> component with the prop inputType set to "radio" should render the <input /> when increaseArea is passed as a negative value 1`] = `
Object {
  "checked": false,
  "disabled": false,
  "label": null,
  "name": null,
  "onBlur": [Function],
  "onChange": [Function],
  "onFocus": [Function],
  "style": Object {
    "background": "#fff",
    "border": 0,
    "cursor": "pointer",
    "display": "block",
    "height": "40%",
    "left": "30%",
    "margin": 0,
    "opacity": 0,
    "padding": 0,
    "position": "absolute",
    "top": "30%",
    "width": "40%",
  },
  "type": "radio",
  "value": "",
}
`;

exports[`<Input /> component with the prop inputType set to "radio" should render the <input /> when increaseArea is passed as a negative value and more than 50% 1`] = `
<div
  className="iradio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "0%",
        "left": "50%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "50%",
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "radio" should render the <input /> when increaseArea is passed as a positive value 1`] = `
<div
  className="iradio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "160%",
        "left": "-30%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "-30%",
        "width": "160%",
      }
    }
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "radio" should render the corresponding HTML when the insert value is set 1`] = `
<div
  class="iradio"
>
  <input
    style="opacity:0;position:absolute"
    type="radio"
    value=""
  />
  <div>
    test
  </div>
  <ins
    class="iCheck-helper"
    role="presentation"
    style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
  />
</div>
`;

exports[`<Input /> component with the prop inputType set to "radio" should render wrapper with aria labels  when aria prop is passed 1`] = `
<div
  aria-checked={false}
  className="iradio"
  role="radio"
>
  <input
    checked={false}
    disabled={false}
    label={null}
    name={null}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
    type="radio"
    value=""
  />
  <ins
    className="iCheck-helper"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "background": "#fff",
        "border": 0,
        "cursor": "pointer",
        "display": "block",
        "height": "100%",
        "left": "0%",
        "margin": 0,
        "opacity": 0,
        "padding": 0,
        "position": "absolute",
        "top": "0%",
        "width": "100%",
      }
    }
  />
</div>
`;
